src/modules/sdexec/start.c

```diff
@@ -0,0 +1,235 @@
+ */
+static int prop_add_env (json_t *prop, const char *name, json_t *dict)
+{
+    json_t *a;
+    const char *k;
+    json_t *vo;
+    json_t *env;
+    int rc = -1;
+
+    if (!(a = json_array ()))
+        return -1;
+    json_object_foreach (dict, k, vo) {
+        const char *v = json_string_value (vo);
+        char *kv = NULL;
+        json_t *kvo = NULL;
+
+        if (asprintf (&kv, "%s=%s", k, v) < 0
+            || !(kvo = json_string (kv))
+            || json_array_append_new (a, kvo) < 0) {
+            json_decref (kvo);
+            free (kv);
+            goto out;
+        }
+        free (kv);
+    }
+    if (!(env = json_pack ("[s[sO]]", name, "as", a))
+        || json_array_append_new (prop, env) < 0) {
+        json_decref (env);
+        goto out;
+    }
+    rc = 0;
+out:
+    json_decref (a);
+    return rc;
+}
+
+static int prop_add_string (json_t *prop, const char *name, const char *val)
+{
+    json_t *o;
+
+    if (val) {
+        if (!(o = json_pack ("[s[ss]]", name, "s", val))
+            || json_array_append_new (prop, o) < 0) {
+            json_decref (o);
+            return -1;
+        }
+    }
+    return 0;
+}
+
+/* This assumes message source and destination are in the same process,
+ * as is the case with sdexec => sdbus broker modules.
+ */
+static int prop_add_fd (json_t *prop, const char *name, int val)
+{
+    json_t *o;
+
+    if (val >= 0) {
+        if (!(o = json_pack ("[s[si]]", name, "h", val))
+            || json_array_append_new (prop, o) < 0) {
+            json_decref (o);
+            return -1;
+        }
+    }
+    return 0;
+}
+
+static int prop_add_bool (json_t *prop, const char *name, int val)
+{
+    json_t *o;
+
+    if (!(o = json_pack ("[s[sb]]", name, "b", val))
+        || json_array_append_new (prop, o) < 0) {
+        json_decref (o);
+        return -1;
+    }
+    return 0;
+}
+
+static json_t *prop_create (json_t *cmd,
+                            const char *desc,
+                            int stdin_fd,
+                            int stdout_fd,
+                            int stderr_fd,
+                            flux_error_t *error)
+{
+    json_t *prop;
+    json_error_t jerror;
+    const char *cwd = NULL;
+    json_t *cmdline;
+    json_t *env;
+
+    // Not unpacked: channels, opts
+    if (json_unpack_ex (cmd,
+                        &jerror,
+                        0,
+                        "{s?s s:o s:o}",
+                        "cwd", &cwd,
+                        "cmdline", &cmdline,
+                        "env", &env) < 0
+        || json_array_size (cmdline) == 0) {
+        errprintf (error, "error parsing command object: %s", jerror.text);
+        errno = EPROTO;
+        return NULL;
+    }
+    if (!(prop = json_array ())) {
+        errprintf (error, "out of memory");
+        errno = ENOMEM;
+        return NULL;
+    }
+    if (prop_add_execstart (prop, "ExecStart", cmdline) < 0
+        || prop_add_string (prop, "Type", "exec") < 0
```

Comment:
just curious where you found "type=exec" documented?  Or dug up in systemd code?  Struggling to figure out where this field comes from, b/c I didn't do this in `libsdprocess`.

Reply:
The various types are described [here](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=)

Reply:
ahhh i had totally forgotten that this stuff is strewn about various manpages too :-)   Perhaps useful to add documentation pointers when you can recall where they came from.  I probably should have done a better job of that in libsdprocess too.

Reply:
No worries - you got it working!

Maybe we could add a markdown page in the sdbus source that includes references and (eventually) some documentation?   I would imagine someone who was not familiar at all with dbus or systemd would find this all pretty mystifying.

Reply:
That sounds like a good idea.  I remember having difficulty even remembering where I found things documented, as little bits are found all over the place :-)